portal/templates/contract/components/issue-or-delete-facility-link.njk

```diff
@@ -1,37 +1,33 @@
-  {% set userIsChecker = params.user.roles.includes('checker')%}
-
+  {% set userIsMaker = params.user.roles.includes('maker') %}
+  {% set userIsChecker = params.user.roles.includes('checker') %}
   {% set dealId = params.deal._id %}
   {% set dealStatus = params.deal.status %}
   {% set dealSubmissionDate = params.deal.details.submissionDate %}
   {% set dealIsEditable = params.editable %}
-
   {% set facility = params.facility %}
   {% set facilityName = params.facilityName %}
   {% set facilityTableIndex = params.facilityTableIndex %}
-
   {% set canOnlyViewIssueFacilityDetails = facility.status === "Submitted" or facility.status === "Ready for check" or facility.status === "Acknowledged" or dealStatus === "Ready for Checker's approval" %}
-
-  {% if facility.canIssueOrEditIssueFacility %}
-    <a href="/contract/{{ dealId }}/{{ facilityName }}/{{ facility._id}}/issue-facility" class="govuk-link" data-cy="{{ facilityName }}-issue-facility-{{ facility._id }}">
-
-    {% if facility.issueFacilityDetailsStarted and facility.issueFacilityDetailsProvided %}
-      Facility issued
+  {% if facility.canIssueOrEditIssueFacility and userIsMaker %}
+    <a href="/contract/{{ dealId }}/{{ facilityName }}/{{ facility._id }}/issue-facility" class="govuk-link" data-cy="{{ facilityName }}-issue-facility-{{ facility._id }}">
+      {% if facility.issueFacilityDetailsStarted and facility.issueFacilityDetailsProvided %}
+        Facility issued
       {% else %}
-      Issue facility
-    {% endif%}
+        Issue facility
+      {% endif %}
     </a>
-
-    {% elseif canOnlyViewIssueFacilityDetails and facility.issueFacilityDetailsProvided %}
+    {# 2 #}
+  {% elseif canOnlyViewIssueFacilityDetails or facility.canIssueOrEditIssueFacility %}
+    {% if facility.issueFacilityDetailsProvided %}
       <a href="/contract/{{ dealId }}/submission-details#{{ facilityName }}-{{ facility._id }}" class="govuk-link" data-cy="{{ facilityName }}-issue-facility-{{ facility._id }}">Facility issued</a>
-
-    {% elseif (userIsMaker and dealIsEditable) and (dealStatus === "Draft" or dealStatus === "Further Maker's input required") and not dealSubmissionDate %}
-      {% if facility.name %}
-        <a href="/contract/{{ dealId }}/{{ facilityName }}/{{ facility._id}}/delete" class="govuk-link" data-cy="{{ facilityName }}-delete-{{ facility._id }}" aria-label="Delete {{ facility.type }} {{ facility.name }}">Delete {{ facility.name }}</a>
-      {% else %}
-        <a href="/contract/{{ dealId }}/{{ facilityName }}/{{ facility._id}}/delete" class="govuk-link" data-cy="{{ facilityName }}-delete-{{ facility._id }}" aria-label="Delete {{ facility.type }} {{ facilityTableIndex }}">Delete {{ facility.type | lower | safe }}</a>
-      {% endif %}
+    {% endif %}
+    {# 3 #}
+  {% elseif (userIsMaker and dealIsEditable) and (dealStatus === "Draft" or dealStatus === "Further Maker's input required") and not dealSubmissionDate %}
```

Comment:
Cognitive complexity is over 3, can this please be feed from the controller, any thoughts?

Reply:
I think this issue is bigger than this ticket -- we have a lot of complex values being calculated inside nunjucks components -- I'd like to have a think about the best way to capture these (I'm sure there's a lot of overlap with these too).